capellambse/model/diagram.py

```diff
@@ -134,13 +134,57 @@ def _short_html_(self) -> markupsafe.Markup:
             f"(uuid: {markupsafe.Markup.escape(self.uuid)})"
         )
 
-    def _repr_svg_(self) -> t.Any | None:
-        if self._model.jupyter_untrusted:
+    def _repr_mimebundle_(
+        self,
+        include: cabc.Container[str] | None = None,
+        exclude: cabc.Container[str] | None = None,
+        **_,
+    ) -> tuple[dict[str, t.Any], dict[t.Any, t.Any]] | dict[str, t.Any] | None:
+        if include is None:
+            include = helpers.EverythingContainer()
+        if exclude is None:
+            exclude = ()
+
+        formats: dict[str, DiagramConverter] = {}
+        eligible_formats = [("image/png", "png")]
+        # XXX Hack to disable SVG in Jupyter notebooks, when requested
+        if not self._model.jupyter_untrusted:
+            eligible_formats.append(("image/svg+xml", "svg"))
```

Comment:
Why is it a hack? Is `_repr_mimebundle_` also not an internal method or appearing in the documentation?

Reply:
![grafik](https://user-images.githubusercontent.com/1579756/180395654-b8e815e6-8e0a-442b-918a-81804220bdc8.png)

`_repr_mimebundle_` is a method [from the IPython kernel](https://ipython.readthedocs.io/en/stable/config/integrating.html#MyObject._repr_mimebundle_) (the docu sucks, but here's an [example notebook](https://github.com/ipython/ipython/blob/master/examples/IPython%20Kernel/Custom%20Display%20Logic.ipynb) that elaborates a bit more).

We simply implement this method, so that Jupyter displays the diagrams nicely without needing any special function calls around them (i.e. so you can just say `diagram` in a cell instead of e.g. `SVG(diagram.as_svg)`).

Before this, we had the `_repr_svg_` and `_repr_png_` methods, which did essentially the same thing, albeit in a slightly different order than `_repr_mimebundle_` does now. And the `_repr_svg_` was "hacked" the same way to disable it when `jupyter_untrusted` was set, it just wasn't explicitly marked as being a hack like it is now.